prow/github/client.go

```diff
@@ -146,9 +147,33 @@ func (t *throttler) Wait() {
 	}
 }
 
+func (t *throttler) Refund() {
```

Comment:
IMO this and the go-routine in `Throttle()` should use the same behavior for refilling

Reply:
I'd be happy to DRY this out if you have any suggestions on how to do that without deadlock.
I thought about moving the switch statement into a shared function parameterized by a channel, but that only DRYs out a few lines and doesn't seem to add much value since acquiring the lock and reading the channel field can't be in the shared function.